### PR TITLE
Update of POD method

### PR DESCRIPTION
**Pull request by zfutera**
Dear Developers,

I'm submitting an update of POD method (PROPERTIES -> ET_COUPLING -> PROJECTION) for electronic-coupling calculations that I implemented in CP2K previously. 

The update allows saving restart files during the POD method (projected Hamiltonian, localized wavefunctions, transformation matrices) and reading the restart files from previous runs. This functionality is useful in works with large systems where the POD transformations takes a lot of time. 

The update is result of my usage of POD work on various systems when I found the restart functionality very useful.

As a part of the pull request, I'm also sending the updated regtests (QS/regtest-pod).

Log of the full regtest run with the modified CP2K version is attached:
[cp2k-pod-update.regtest.log](https://github.com/cp2k/cp2k/files/4744683/cp2k-pod-update.regtest.log)

Kind regards,

Zdenek Futera

### CODE REVIEW

**Comment by oschuett** [comment]
Thanks for you PR. The formatting test is failing. Can you please run `make -j pretty`.

**Comment by zfutera** [comment]
Sorry for that, I forgot to check prettifying. That should be OK now.

**Comment by zfutera** [comment]
You right, my fault. OK, I'll fix it. Thanks

**Comment by zfutera** [comment]
Sorry, I've got a bit lost. I typed those commands (reset, fetch an rebase) but now I'm getting this message:

> Your branch and 'origin/master' have diverged,
> and have 1 and 4 different commits each, respectively.
>   (use "git pull" to merge the remote branch into yours)

What should I do next? Pull?

**Comment by dev-zero** [comment]
> Sorry, I've got a bit lost. I typed those commands (reset, fetch an rebase) but now I'm getting this message:
> 
> > Your branch and 'origin/master' have diverged,
> > and have 1 and 4 different commits each, respectively.
> > (use "git pull" to merge the remote branch into yours)
> 
> What should I do next? Pull?

Now you can do a `git push --force` which brings also the remote-copy of your branch back to the state before the merge commit.

**Comment by zfutera** [comment]
Done. Thanks!

**Comment by oschuett** [comment]
Oh, I forgot. Now you'll need `git push --force`.

That being said, you could have just continued to use `commit` and `push`. The `fetch` / `rebase` / `push --force` is only needed when someone else has committed conflicting changes in the meantime. In that case `rebase` gives you the opportunities to resolve those conflicts.

**Comment by zfutera** [comment]
Oh, it seems that I forgot remove some unused routine arguments from the code. Sorry. I'll try to fix it.

File: src/et_coupling_pod_utils.F [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+      ! Forward ransformations S^{-1/2}
+      IF (output_unit > 0) &
+         WRITE (output_unit, '(/,T3,A)') &
+         'Constructing forward basis-set transformation matrix'
+
+      DO i = 1, n_ao
+         IF (et(i) >= scf_cntrl%eps_eigval) THEN
+            f = et(i)**(-0.25_dp)
+#if defined(__SCALAPACK)
```
Comment (oschuett) [inline src/et_coupling_pod_utils.F]: Please use [cp_fm_set_all](https://github.com/cp2k/cp2k/blob/master/src/fm/cp_fm_types.F#L365) instead. Generally, high level code like this should not contain any `#if defined`.

Comment (zfutera) [reply src/et_coupling_pod_utils.F]: OK, I'll look at it, thanks!

**Comment by zfutera** [comment]
I checked the code and removed some of the `#if defined` directives, however, in some cases I didn't find an alternative and had to keep them in the code (these parts are related to reordering elements of large matrices or printing of selected elements)

I've also modified one of the regtests because its run was too long.

Could you please review it and let me know whether any other improvements are needed? Thanks!

**Comment by oschuett** [comment]
> in some cases I didn't find an alternative and had to keep them in the code (these parts are related to reordering elements of large matrices or printing of selected elements)

Could you maybe add some routines for reordering and selecting elements to the fm library? After all, the main purpose of  that library is to encapsulate the differences between parallel and serial builds.

>  any other improvements are needed?

Your code looks otherwise very good. However, I noticed your tests are taking quite a long time. Most of our tests run within a few seconds and [30 seconds](https://github.com/cp2k/cp2k/issues/174) should be the upper limit. Could you please look into that?

**Comment by zfutera** [comment]
OK, I can do that, thank's for suggestions!

**Comment by oschuett** [comment]
Oh, why did you close this PR? Do you need help with adding those routines to the fm library?

**Pull request closed without merging**